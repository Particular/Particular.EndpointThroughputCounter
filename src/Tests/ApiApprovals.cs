namespace Tests;

using NUnit.Framework;
using Particular.Approvals;
using Particular.EndpointThroughputCounter.Data;
using PublicApiGenerator;

[TestFixture]
public class APIApprovals
{
    [Test]
    public void Approve()
    {
        var publicApi = typeof(Report).Assembly.GeneratePublicApi(new ApiGeneratorOptions
        {
            ExcludeAttributes = new[] { "System.Runtime.Versioning.TargetFrameworkAttribute", "System.Reflection.AssemblyMetadataAttribute" }
        });
        Approver.Verify(publicApi);
    }
}