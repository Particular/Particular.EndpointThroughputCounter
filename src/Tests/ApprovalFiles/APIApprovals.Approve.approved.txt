namespace Particular.EndpointThroughputCounter.Data
{
    public class QueueThroughput
    {
        public QueueThroughput() { }
        [Newtonsoft.Json.JsonProperty(DefaultValueHandling=Newtonsoft.Json.DefaultValueHandling.Include | Newtonsoft.Json.DefaultValueHandling.Ignore)]
        public bool NoDataOrSendOnly { get; set; }
        public string QueueName { get; set; }
        [Newtonsoft.Json.JsonProperty(DefaultValueHandling=Newtonsoft.Json.DefaultValueHandling.Include | Newtonsoft.Json.DefaultValueHandling.Ignore)]
        public int? Throughput { get; set; }
    }
    public class Report
    {
        public Report() { }
        public string CustomerName { get; set; }
        public System.DateTimeOffset EndTime { get; set; }
        [Newtonsoft.Json.JsonProperty(DefaultValueHandling=Newtonsoft.Json.DefaultValueHandling.Include | Newtonsoft.Json.DefaultValueHandling.Ignore)]
        public string[] IgnoredQueues { get; set; }
        public string MessageTransport { get; set; }
        [Newtonsoft.Json.JsonProperty(DefaultValueHandling=Newtonsoft.Json.DefaultValueHandling.Include | Newtonsoft.Json.DefaultValueHandling.Ignore)]
        public string Prefix { get; set; }
        public Particular.EndpointThroughputCounter.Data.QueueThroughput[] Queues { get; set; }
        public System.TimeSpan ReportDuration { get; set; }
        public string ReportMethod { get; set; }
        public System.DateTimeOffset StartTime { get; set; }
        public string ToolVersion { get; set; }
        public int TotalQueues { get; set; }
        public int TotalThroughput { get; set; }
    }
    public class SignedReport
    {
        public SignedReport() { }
        public Particular.EndpointThroughputCounter.Data.Report ReportData { get; set; }
        public string Signature { get; set; }
    }
}